addons/website_slides/models/slide_slide.py

```diff
@@ -877,16 +911,39 @@ def _action_set_completed(self, target_partner):
             'vote': 0,
             'completed': True} for new_slide in new_slides])
 
-        return True
+    def action_mark_uncompleted(self):
+        if any(not slide.can_self_mark_uncompleted for slide in self):
+            raise UserError(_('You cannot mark a slide as uncompleted if you are not among its members.'))
 
-    def _action_set_quiz_done(self):
-        if any(not slide.channel_id.is_member for slide in self):
-            raise UserError(_('You cannot mark a slide quiz as completed if you are not among its members.'))
+        if any(not slide.user_has_completed for slide in self):
+            raise UserError(_('This slide is already set as uncompleted'))
```

Comment:
Same remark as other method, we should probably just filter them out.
That way marking complete / incomplete slides that are already in that state will simply do nothing.

(That will also avoid errors in case there is some lag in the server and the end-user manages to click multiple times on the button before we complete the rpc).